operators/src/mock/mock_dataset_data_source.rs

```diff
@@ -130,6 +131,10 @@ impl VectorOperator for MockDatasetDataSource {
         }
         .boxed())
     }
+
+    fn span(&self) -> CreateSpan {
+        || span!(Level::TRACE, MockDatasetDataSource::TYPE_NAME)
+    }
```

Comment:
Is there any way we can use the `OperatorName` trait for that?

Reply:
I do use the `OperatorName` trait, as the  `TYPE_NAME` field is part of it. However, like I tried to explain in the Trello card, we can't use the trait as a generic. We can only input the `TYPE_NAME` of a concrete type. Otherwise Rust complains on not being able to caputre the outer generic of a function. This has to do with the way the `span!` macro works, as far as I undrstand.

Reply:
Is this the issue?
https://github.com/tokio-rs/tracing/issues/1047

Reply:
no, the `'static` is not the problem, but the string having to be `const` in order for the macro to work